src/nidigital/templates/session.py/send_software_edge_trigger.py.mako

```diff
@@ -0,0 +1,34 @@
+<%page args="f, config"/>\
```

Comment:
An alternative aproach so we duplicate less lower level code would be to let the generator generate `_send_software_edge_trigger` from metadata but as private. We could even have the enum for selecting the trigger which _should_ be in metadata.

Then the fancy wrapper calls `_send_software_edge_trigger`.

Reply:
If we go with the solution of using enum values to differentiate between different kinds of trigger, we would probably need a new python-only enum in nidigital because the [existing enum](https://github.com/ni/nimi-python/blob/f047438c6f40b93f9e2377d5aee806d5d3068b99/src/nidigital/metadata/enums.py#L350) is named `Signal` (which is weird in the context of `send_software_edge_trigger`) and it has values that cannot be passed to `send_software_edge_trigger` - NIDIGITAL_VAL_PATTERN_OPCODE_EVENT and NIDIGITAL_VAL_REF_CLOCK.

The reason for this mess is the use of same constants between `send_software_edge_trigger` and `export_signal` in C API. In python API, we got rid of `export_signal`. So we could probably give the enum a more appropriate name and get rid of the values that are not needed.

I think this is not a deal-breaker for the "enums" solution, just something extra to do.